src/all/pandachaika/src/eu/kanade/tachiyomi/extension/all/pandachaika/PandaChaika.kt

```diff
@@ -0,0 +1,326 @@
+package eu.kanade.tachiyomi.extension.all.pandachaika
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Interceptor
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.Protocol
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import uy.kohesive.injekt.injectLazy
+import java.lang.String.CASE_INSENSITIVE_ORDER
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+class PandaChaika(
+    override val lang: String = "all",
+    private val searchLang: String = "",
+) : HttpSource() {
+
+    override val name = "PandaChaika"
+
+    override val baseUrl = "https://panda.chaika.moe"
+
+    private val baseApiUrl = "$baseUrl/api"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient
+        .newBuilder()
+        .addInterceptor(::Intercept)
+        .build()
+
+    private val json: Json by injectLazy()
+
+    // Popular
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/search/?tags=$searchLang&sort=rating&apply=&json=&page=$page", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val library = response.parseAs<ArchiveResponse>()
+
+        val mangas = library.archives.map(::popularManga)
+
+        val hasNextPage = library.has_next
+
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    private fun popularManga(hentai: LongArchive) = SManga.create().apply {
```

Comment:
all the info seems to be availble in `/search/` call so parse everything there, and add `initialized = true` to smanga in order to avoid one extra network call for fetching details